xarray/tests/test_interp.py

```diff
@@ -462,19 +462,31 @@ def test_interp_like():
 
 
 @requires_scipy
-def test_datetime():
-    da = xr.DataArray(np.random.randn(24), dims='time',
+@pytest.mark.parametrize('x_new, expected', [
+    (pd.date_range('2000-01-02', periods=3), [1, 2, 3]),
+    (np.array([np.datetime64('2000-01-01T12:00'),
+               np.datetime64('2000-01-02T12:00')]), [0.5, 1.5]),
+    (['2000-01-01T12:00', '2000-01-02T12:00'], [0.5, 1.5]),
+    (['2000-01-01T12:00'], 0.5),
+    pytest.param('2000-01-01T12:00', 0.5, marks=pytest.mark.xfail)
+])
+def test_datetime(x_new, expected):
+    da = xr.DataArray(np.arange(24), dims='time',
                       coords={'time': pd.date_range('2000-01-01', periods=24)})
 
-    x_new = pd.date_range('2000-01-02', periods=3)
     actual = da.interp(time=x_new)
-    expected = da.isel(time=[1, 2, 3])
-    assert_allclose(actual, expected)
+    expected_da = xr.DataArray(np.atleast_1d(expected), dims=['time'],
+                               coords={'time': (np.atleast_1d(x_new)
+                                                .astype('datetime64[ns]'))})
 
-    x_new = np.array([np.datetime64('2000-01-01T12:00'),
-                      np.datetime64('2000-01-02T12:00')])
-    actual = da.interp(time=x_new)
-    assert_allclose(actual.isel(time=0).drop('time'),
-                    0.5 * (da.isel(time=0) + da.isel(time=1)))
-    assert_allclose(actual.isel(time=1).drop('time'),
-                    0.5 * (da.isel(time=1) + da.isel(time=2)))
+    assert_allclose(actual, expected_da)
+
+
+@requires_scipy
+def test_datetime_single_string():
```

Comment:
@shoyer the scalar case works and is tested here. I didn't want to make the previous test too complicated (it fails because the result of the scalar case has no dims; but the test checks both dims and coords).

Reply:
OK, cool :)